qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -200,26 +194,23 @@ def aux_operators(self, aux_operators: List[BaseOperator]) -> None:
         self._in_aux_operators = aux_operators
 
     @VQAlgorithm.var_form.setter
-    def var_form(self, var_form: VariationalForm):
+    def var_form(self, var_form: Union[QuantumCircuit, VariationalForm]):
         """ Sets variational form """
         VQAlgorithm.var_form.fset(self, var_form)
-        self._var_form_params = ParameterVector('θ', var_form.num_parameters)
-        if self.initial_point is None:
+
+        if self.initial_point is None and hasattr(var_form, 'preferred_init_points'):
             self.initial_point = var_form.preferred_init_points
         self._check_operator_varform()
 
     def _check_operator_varform(self):
+        """Check that the number of qubits of operator and variational form match."""
         if self.operator is not None and self.var_form is not None:
             if self.operator.num_qubits != self.var_form.num_qubits:
-                # TODO After Ansatz update we should be able to set in the
-                #      number of qubits to var form. Important since use by
-                #      application stack of VQE the user may be able to set
-                #      a var form but not know num_qubits. Whether any smarter
-                #      settings could be optionally done by VQE e.g adjust depth
-                #      is TBD. Also this auto adjusting might not be reasonable for
-                #      instance UCCSD where its parameterization is much closer to
-                #      the specific problem and hence to the operator
-                raise AquaError("Variational form num qubits does not match operator")
+                # try to set the number of qubits on the variational form, if possible
+                try:
+                    self.var_form.num_qubits = self.operator.num_qubits
+                except AttributeError:
+                    raise AquaError("Variational form num qubits does not match operator")
```

Comment:
The error seems more like the var form does not have such an attribute. The circuit and variational form should have so is this more case of trying to handle being given something that doesn't and less about matching. If a user passed in a var form with num qubits already set we would just override that based on the operator here right. For UCCSD to work here without failure it must continue to have num qubits as a settable.

Reply:
So I should add to the error message, that the number of qubits is not settable, right?

Reply:
I would think so if we want to check. The only two objects that were supposed to be passed as a var form ie QuantumCircuit and Var Form both have this as an attribute hence this check is really unneccessary. Some code was there evidently anticipating setting num qubits - which was not on the orginal var forms.

Reply:
The circuit in general does not allow setting the number of qubits (at the moment, it is being discussed if that should be changed!) -- so we need to check. I changed the msg so that it says the number of qubits cannot be changed, additionally.

Reply:
Ok, I guess we never ended up with any subclass say like a LibraryCircuit where this was always supported. Having to check whether it has attributes or not is not so great but I guess it is what it is  for now